src/class_loader.cpp

```diff
@@ -150,12 +150,11 @@ int ClassLoader::unloadLibraryInternal(bool lock_plugin_ref_count)
 
   try {
     if (plugin_ref_count_ > 0) {
-      CONSOLE_BRIDGE_logWarn(
-        "%s",
-        "class_loader.ClassLoader: SEVERE WARNING!!! "
-        "Attempting to unload library while objects created by this loader exist in the heap! "
-        "You should delete your objects before attempting to unload the library or destroying "
-        "the ClassLoader. The library will NOT be unloaded.");
+      CONSOLE_BRIDGE_logWarn("class_loader.ClassLoader: SEVERE WARNING!!!\n"
+        "Attempting to unload %s\n"
+        "while objects created by this library still exist in the heap!\n"
+        "You should delete your objects before destroying the ClassLoader. "
+        "The library will NOT be unloaded.", library_path_.c_str());
```

Comment:
In most cases is this a side-affect of the ClassLoader object going out of scope and being destroyed while the plugin(s) it has loaded are still in scope?

Would it be helpful to add this to the error message?

Please verify that the lifetime of the ClassLoader for {} is at least as long as the lifetime of all plugins loaded by that ClassLoader.